chain/chain/src/resharding.rs

```diff
@@ -248,7 +270,9 @@ impl Chain {
         // The snapshot when created has the flat head as of `prev_prev_hash`, i.e. the hash as
         // of the second last block of the previous epoch. Hence we need to append the detla
         // changes on top of it.
-        let (snapshot_store, flat_storage_manager) = tries.get_state_snapshot(&prev_prev_hash)?;
+        let (snapshot_store, flat_storage_manager) = tries
+            .get_state_snapshot(&prev_prev_hash)
```

Comment:
I think it should be `prev_hash`, the snapshot is triggered after processing a block about which we know that the next block will start a new epoch.

Reply:
This has been tested to work :) 

I believe capturing the snapshot is triggered even a bit later, after processing the first block of the epoch or so. It kinda makes sense to me as I was a bit worried about forks and whatnot. Shreyan knows this better than I but I think we set the head and name of the snapshot to the prev_prev_hash but we still have delta so that we can always access data as of prev_hash.